_posts/2022-09-06-conscryptmode.md

```diff
@@ -0,0 +1,13 @@
+GSoC is an open-source program where participants/student developers can contribute to a project from a set of different organizations while learning new things and they get some monetary benefits from Google on completion of their work.
+
+My project was `Switching Robolectric from BouncyCastle to Conscrypt as the default security provider`. Robolectric was using BouncyCastle as the Java Cryptography Extension (JCE) security provider. But after the introduction of Android P, Android switched to using Google Conscrypt as the security provider. 
+To be more consistent with Android, Robolectric should be updated the default security provider to Conscrypt.
+ 
+I am already well familiar with the technologies used in the project such as Java, Kotlin, Android Studio, Git and GitHub. I am interested in Android Developement and I was looking for an android project that uses Java as the primary language. I found Robolectric a perfect project to contribute according to my skills and I was ready to learn about testing and other related stuff as Robolectric is an Android Unit Testing Framework. 
+
+First we thought to switch Robolectric from BouncyCastle to Conscrypt as default security provider. But soon we found we can't completely switch Robolectric to Conscrypt because there were some internal support issues that need BouncyCastle as fallback security provider. After that we decided to create a ConscryptMode annotation that helps us to either make use of Conscrypt or not to use it. Robolectric can currently either use Conscrypt and BouncyCastle or just BouncyCastle as the security provider.
+
+- If ConscryptMode is `ON`, it will install Conscrypt and BouncyCastle.
+- If ConscryptMode is `OFF`, it will only install BouncyCastle.
+
```

Comment:
This would be a good place to talk about what specific programming things you learned about or worked with too, for example:

1. Unit testing best practices.
2. Test state being retained between test methods and all the problems that can cause.
3. Debugging skills
4. Annotations/Modes
5. How Robolectric works
6. Contributing to an open source repo, dealing with internal vs external repos.